syncd/Syncd.cpp

```diff
@@ -5103,7 +5112,7 @@ bool Syncd::isVeryFirstRun()
      * this is first run, let's query HIDDEN ?
      */
 
-    bool firstRun = m_client->hasNoHiddenKeysDefined();
+    bool firstRun = m_client->hasNoHiddenKeysDefined() && m_client->hasNoSwitchDefined();
```

Comment:
You already have flag isveryfirst run, why this is modified?

Reply:
firstRun being true will disallow the start of warm-reboot.
It was set to true when no switch attributes is found under the key of HIDDEN* in ASIC_DB or GB_ASIC_DB. However, all such attributes under HIDDEN are unrelated to the credo gbsyncd, so hasNoHiddenKeysDefined is always true for gbsyncd.
The change here is to determine the first run also based on no switch objects has been created in ASIC_DB or GB_ASIC_DB.